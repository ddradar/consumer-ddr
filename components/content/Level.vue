<template>
  <div class="tags has-addons">
    <NuxtLink
      :to="`/series/${series.slug}/`"
      class="tag"
      :style="{ backgroundColor: series.color, color: '#fff' }"
    >
      <span :title="series.title">{{ series.slug }}</span>
    </NuxtLink>
    <span class="tag">{{ level }}</span>
  </div>
</template>

<script lang="ts" setup>
import type { Song } from '~~/src/content'

const NuxtLink = resolveComponent('NuxtLink')

interface LevelProps {
  series: Song['series'][string]
  level: number | '10+' | '?'
}

const props = defineProps<LevelProps>()
</script>
