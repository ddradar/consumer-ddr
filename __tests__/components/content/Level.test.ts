import { mount, RouterLinkStub } from '@vue/test-utils'
import { describe, expect, test, vi } from 'vitest'

import Level from '~~/components/content/Level.vue'

vi.mocked(resolveComponent).mockReturnValue(RouterLinkStub as any)

describe('components/content/Level.vue', () => {
  test('renders series link', () => {
    const wrapper = mount(Level, { props: { slug: '1st-jp', level: 5 } })
    expect(wrapper.element).toMatchSnapshot()
  })
})
