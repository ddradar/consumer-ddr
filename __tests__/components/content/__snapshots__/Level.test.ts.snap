// Vitest Snapshot v1

exports[`components/content/Level.vue > renders series link 1`] = `
<div
  class="tags has-addons"
>
  <a
    class="tag"
    style="background-color: darkviolet; color: rgb(255, 255, 255);"
  >
    <span
      title="Dance Dance Revolution (JP)"
    >
      1st-jp
    </span>
  </a>
  <span
    class="tag"
  >
    5
  </span>
</div>
`;
